model/lifecycle.go

```diff
@@ -348,16 +345,15 @@ func RestartBuild(ctx context.Context, b *build.Build, taskIDs []string, abortIn
 	if err != nil {
 		return errors.Wrap(err, "finding completed tasks for build")
 	}
-	if len(completedTasks) == 0 {
-		return nil
-	}
-
 	return restartTasks(ctx, completedTasks, caller, b.Version)
 }
 
 // restartTasks restarts all finished tasks in the given list that are not part of
 // a single host task group.
 func restartTasks(ctx context.Context, allFinishedTasks []task.Task, caller, versionId string) error {
+	if len(allFinishedTasks) == 0 {
```

Comment:
This is a consolidation of the two `if len(completedTasks) == 0 {` instances above